README.md

```diff
@@ -0,0 +1,163 @@
+  setup, bidding, and settlement.
+
+### Technology Stack
+
+- **Smart Contract**: Solidity
+- **Circuit**: Rust, SP1
+- **Encryption scheme**: secp256k1, AES-256-GCM, HKDF-SHA256
+- **Verifier**: Aligned layer
+
+### Workflow
+
+The core logic of **Tahm-Kech** operates on-chain, while off-chain processes handle the winner calculation and proof
+generation. The auction process follows four main phases:
+
+1. **Initial Setup**: The auction owner creates the auction, sets the required deposit amount, transfers assets to the
+   smart contract,
+   and defines the auction's start and end times.
+2. **Bid Phase**: Bidders submit their bids to the smart contract by depositing the required amount.
+3. **Open Phase**: After the bidding window closes, the auction owner calculates the winner and generates a
+   zero-knowledge proof.
+4. **Verify Phase**: The owner submits the winner and the proof to the smart contract for verification, concluding the
+   auction.
+
+**Sequence diagrams for each module are shown below:**
+
+#### Auction Flow:
+
+![auction_flow](diagram/auction_flow.png)
+
+#### Prover & Verifier Flow:
+
+![prover_verifier_flow](diagram/prover_verifier_flow.png)
+
+### Challenges
+
+We used the [ecies](https://crates.io/crates/ecies) crate to encrypt and decrypt bids. However, the proving time and
+proof size were substantial: it took **6 minutes** to generate a proof for **2 bids**, resulting in a **21MB** proof
+size on a system with **64GB RAM** and an **i5-13500 CPU**. This large proof size couldn't be submitted to the Aligned
+layer, so we compressed it, which further increased the proving time.
+
+To address this, we rebuilt the encryption scheme using SP1's patched `secp256k1` crate. This reduced the proving time
+by **half** and compressed the proof size to **1.5MB**. However, the proof couldn't be verified by SP1 due to the
+error `Core(Invalid shard proof: Out-of-domain evaluation mismatch on chip CPU)`. You can review the code in
+the [feat/ecies](https://github.com/sota-zk-labs/tahm-kench/tree/feat/ecies) branch.
+
+### Future Plans
+
+SP1's performance limitations have led us to plan a custom circuit for the encryption scheme using `Gnark` to optimize
+proving time and proof size. Following this development, we aim to introduce additional auction types, such as **Unique
+Lowest Bid Auctions** and **Dutch Auctions**. We also plan to explore collaborations with other **dApps**, including
+**DeFi platforms**, to facilitate automatic asset management based on auction outcomes.
+
+### Demo
+
+You can watch our demo video [here](https://youtu.be/Mp-65Ou7t04).
+
+## Instructions
+
+### Requirements
+
+Ensure you have the following installed:
+
+1. [Rust](https://www.rust-lang.org/tools/install)
+2. [Foundry](https://getfoundry.sh)
+3. [Aligned CLI](https://docs.alignedlayer.com/introduction/1_try_aligned)
+4. [SP1](https://docs.succinct.xyz/getting-started/install.html)
+
+### Setup
```

Comment:
Reorganize structure of this file

Reply:
I wrote this following the required structure of Aligned
![image](https://github.com/user-attachments/assets/98af5277-926b-4a29-a0c5-f6240d35361a)

Reply:
Where's "Demo", "Instructions" ???

Reply:
Roadmap is the last?